xarray/core/dataset.py

```diff
@@ -1986,12 +1986,12 @@ def _validate_interp_indexers(
                 else:
                     yield k, v
             elif isinstance(v, int):
-                yield k, Variable((), v)
+                yield k, Variable((), v, attrs=self.coords[k].attrs)
```

Comment:
If understand correctly this (and the changes below) will always keep the attrs, regardless of the value of the `keep_attrs` option or a hypothetical `keep_attrs` option to `interp`. They also make the `attrs` propagation a bit more complicated: if a indexer is a `Variable` (or `DataArray`) object its `attrs` will be kept (even if that's a empty dict), but if it wasn't a `xarray` object the `attrs` of the coordinate will be used instead.

I think we want to always take the indexer `attrs` (or drop them) for `bool` values of `keep_attrs`. Once the combination of `keep_attrs` and `combine_attrs` is finished we can find of a way to specify that we want to keep the old `attrs` for all interpolated dimensions.

Reply:
This seems like a much larger change. I'm guessing I would have to add keep_attrs to `interp`, `interpolate_like`, `_validate_interp_indexers`, `_validate_indexers` and add if checks in every yield of these `_validate*indexers` which surely will slow down an already slow function even further. 
I'm not sure I like this because I really don't want to add more code to slog through in the interp functions unless it's necessary. I imagine this flag will never change from `True` in 99.99% of the cases, right? 

Do we feel strongly that this is the way forward?

Reply:
I think copying over `ds.coords["x"].attrs` for `ds.interp(x=[1,2,3])` is sensible. Perhaps we should use the `drop_conflicts` option on attrs from the indexer and attrs from the coordinate?

As for `keep_attrs` we can open an issue to add compatibility later...